composer/loggers/tensorboard_logger.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Log to `Tensorboard <https://www.tensorflow.org/tensorboard/get_started#:~:text=TensorBoard%20is%20a%20tool%20for,dimensional%20space%2C%20and%20much%20more./>`_."""
+
+from pathlib import Path
+from typing import Any, Callable, Dict, Optional
+
+from torch.utils.tensorboard import SummaryWriter
+
+from composer.core.state import State
+from composer.loggers.logger import Logger, LogLevel
+from composer.loggers.logger_destination import LoggerDestination
+from composer.utils import dist
+
+__all__ = ['TensorboardLogger']
+
+
+class TensorboardLogger(LoggerDestination):
+    """Log to `Tensorboard <https://www.tensorflow.org/tensorboard/>`_.
+
+    Args:
+        log_dir (str, optional): The path to the directory to put these logs in. If it is
+            None the logs will be placed in `./runs/{month}{day}{HH-MM-SS}_{device_name}.local
+        artifact_name (str, optional): Format string for the logfile's artifact name.
```

Comment:
Ok I gave it some thought and I decided to reorganize the arguments a little. Here's what I did:

1. expose `run_name` as an argument to the user.
2. expose `log_dir` as the directory that stores all the other directories named after `run_name`.
3. Set the suffix of each log file will be the `run_name`
4. `artifact_name` is abstracted away from the user because there is pretty much no use case where they need to set this and it adds confusion. We then grab the stem of the log file path and set that as the `artifact_name`

Please refer to the docstring for more clarification

Reply:
The log file path is configured by `run_name` and `log_dir` like so: `{log_dir}/{tensorboard_log_file_prefix_that_cannot_be_modified}{run_name}`